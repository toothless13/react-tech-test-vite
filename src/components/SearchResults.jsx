import '../styles/SearchResults.css';

const SearchResults = ({ results }) => {
  let id = 0;
  if(!results.length) {
    return <p>No results!</p>
  } else {
    return (
      <>
        {results.map((result, index) => {
          return <img className='search-result-image' src={result} alt={`search result: ${index}`} data-testid='search-result-image' key={id++} />
        })}
      </>
    );
  }
}

export default SearchResults;