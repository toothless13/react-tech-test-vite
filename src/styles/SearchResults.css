.search-result-image {
  width: 500px;
  padding: 1rem .5rem;
  /* border: 5px solid rgb(252, 61, 33); */
  border-radius: 25px;
}

@media only screen and (max-width: 550px) {
  .search-result-image {
    width: 90%;
  }
}