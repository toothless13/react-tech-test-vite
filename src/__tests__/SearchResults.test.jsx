import { render, screen } from '@testing-library/react';
import SearchResults from '../components/SearchResults';

describe('SearchResults', () => {
  const validProps = { results: ["https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80", "https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"] };

  const { results } = validProps;
  it('renders correctly', () => {
    const { asFragment } = render(<SearchResults results={results} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the image to the screen', () => {
    render(<SearchResults results={results} />);

    expect(screen.getAllByTestId('search-result-image')).toBeTruthy();
  });
});