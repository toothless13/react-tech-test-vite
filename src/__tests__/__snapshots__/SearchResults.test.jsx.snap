// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <img
    alt="search result: 0"
    class="search-result-image"
    data-testid="search-result-image"
    src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
  />
  <img
    alt="search result: 1"
    class="search-result-image"
    data-testid="search-result-image"
    src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
  />
</DocumentFragment>
`;
